src/Halogen/Hooks/Extra/Hooks/UseDebouncer.purs

```diff
@@ -60,38 +65,41 @@ useDebouncer
   => Milliseconds
   -> (a -> HookM m Unit)
   -> Hook m (UseDebouncer a) (a -> HookM m Unit)
-useDebouncer ms fn = Hooks.wrap Hooks.do
-  _ /\ ref <- Hooks.useRef { debounce: Nothing, val: Nothing }
+useDebouncer ms fn = Hooks.wrap hook
+  where
+  hook :: Hook m (UseDebouncer' a) (a -> HookM m Unit)
+  hook = Hooks.do
+    _ /\ ref <- Hooks.useRef { debounce: Nothing, val: Nothing }
```

Comment:
Mind sharing why you decided to drop the `Hooks.wrap Hooks.do` style for `Hooks.wrap hook where ...` style?

Reply:
Hi

This is how thomashoneyman writes custom hooks in examples in hooks repo

But it will work without too, `HookNewtype` will properly find `UseDebouncer'`

Reply:
Ah, ok.

I'd prefer not to follow his example here. The update you're doing is rather small but because it introduces whitespace differences, it causes a huge diff that makes things harder to read. Thomas' examples are probably prioritizing clarity, so doing it that way makes sense in his examples. I don't think this library needs to follow suit.

Reply:
I switched to implementing the body of the hook in a `where` because I find the spans on the type errors are a bit better. But it's certainly not necessary.